### ISSUE TITLE
Folder View?

### ISSUE THREAD

**Issue by Pompey69** [stashapp/stash#1586]
I was wondering if there was a folder view feature in stash?

I have downloads in folders that are already organized. Can I view the scenes by foler saved on the hard drive?

**Comment by Phasetime** [comment]
Currently the only way to view the contents of actual folders is by using the "path" filter and either using an absolute path with the "is" modifier or a relative path with the "includes" modifier. There is no dedicated folder view though. 
By using the aforementioned filter you can quite easily replicate your folder organization inside stash by selecting all scenes that come up in the query and batch tagging them.

**Comment by Pompey69** [comment]
Shame I really like the app-most of my videos are already organized in folders, I was hoping there was some way to use a folder view.

**Comment by secondfolder** [comment]
That’s something I think would be really useful to. You may want to vote for this issue: https://github.com/stashapp/stash/issues/1029

**Comment by WithoutPants** [comment]
I think that this issue is distinct from #1029. My understanding of the scope of this issue is for a filesystem view of content.

**Comment by WithoutPants** [comment]
I think that this feature needs to be done after the file overhaul discussed in #1958. We will also need to implement some additional back-end schema changes to facilitate efficient folder-based views.

There are a bunch of UI-side design decisions that need to be made around this, and at the moment I don't have a design that I am happy with.

Possible ideas:
* a new view button on the list pages. This view would initially show folders for each configured library. Clicking on a folder would filter the results based on items (and folders) in that directory. Subfolders would show at the top of the list, in addition to the items in the directory. 

The main issue with this approach is that we would have to make a decision on what view type (card, list, tree view for example). We would inevitably get requests for other view types incorporating the file view.

* a top-level files view incorporating all media types

This is a possibility, though probably involves the largest amount of effort. It also requires exposing the file concept to the front-end (the current approach is to only expose file concepts via the media items). 

* a hybrid of the file view in the list pages. This might be triggered via a checkbox or filter option or something. When active, the files/folders are shown.

**Comment by DWreck1995** [comment]
> I think that this feature needs to be done after the file overhaul discussed in #1958. We will also need to implement some additional back-end schema changes to facilitate efficient folder-based views.
> 
> There are a bunch of UI-side design decisions that need to be made around this, and at the moment I don't have a design that I am happy with.
> 
> Possible ideas:
> 
> * a new view button on the list pages. This view would initially show folders for each configured library. Clicking on a folder would filter the results based on items (and folders) in that directory. Subfolders would show at the top of the list, in addition to the items in the directory.
> 
> The main issue with this approach is that we would have to make a decision on what view type (card, list, tree view for example). We would inevitably get requests for other view types incorporating the file view.
> 
> * a top-level files view incorporating all media types
> 
> This is a possibility, though probably involves the largest amount of effort. It also requires exposing the file concept to the front-end (the current approach is to only expose file concepts via the media items).
> 
> * a hybrid of the file view in the list pages. This might be triggered via a checkbox or filter option or something. When active, the files/folders are shown.

A similar route to Plex with having the Scenes, Images, Movies, etc... appearing on the left side. Allowing more space to add custom line items if needed. Then when you open one of them, lets say Scenes for example, it populates as it currently does. But along with all the other filtering and sorting options found at the top, there's also a Folder View option that simply repopulates everything with folders organized the same way they appear in the file system.

I think incorporating a Home tab could be useful as well. The Home tab could be utilized for many different things. Options could be added in the interface settings to allow the user to pick which items appear on the home page. Such as, Continue Watching, Recently Played, Favorite Scenes/Folders, Recently Added, etc...

**Comment by newadventure079** [comment]
I agree with having a folder view.  I have a main folder but within that are subfolders where some things are already partially organized.  It'd be nice to see those folders and drill down into them separately, rather than seeing those vids at the same root level as all the other vids.

**Comment by 0xb0af** [comment]
> I think that this feature needs to be done after the file overhaul discussed in #1958. We will also need to implement some additional back-end schema changes to facilitate efficient folder-based views.

Is progress still being made on the file overhaul? Do you still see it as a blocker for a filesystem /folder view?

> There are a bunch of UI-side design decisions that need to be made around this, and at the moment I don't have a design that I am happy with.
 
The way I'd envision this is similar to the Infuse app for iOS. In that app, media that's been recognized gets filed under Movies, as a movie, or under TV, split into Shows / Seasons / Episodes, but then a separate top-level view is an icon view of the root folder of the library.

In stash, I would vote for a similar grid of icons with two types of items - subfolders, and files. Opening a subfolder would replace the current view with the contents of that folder (possibly with a sort of breadcrumbs bar across to the top to retrace your steps if you're several folders deep). 

The other items would be files. Clicking a file would open it in the player.  A key feature would be the ability to select multiple files and edit metadata - add tags, rate with stars, etc - and have the chosen tag(s) apply to every selected item. 

Notably there is no ability to move files around, rename folders etc. This seems out of scope IMHO.

I'm not familiar with the Stash codebase but if it was easier to make the second type of item Media instead of actual Files, I think that would be fine as long the actual filename could still be presented (as it currently is under the File Info tab of the player screen).
